### ISSUE TITLE
Mobile:  Accessibility: Submitting a new tag in the tag association screen by pressing enter with an external keyboard adds a partial matching tag as well as the new tag

### ISSUE THREAD

**Issue by mrjo118** [laurent22/joplin#12952]
### Operating system

Android

### Joplin version

3.4.4

### Current behaviour

Submitting a new tag in the tag association screen by pressing enter with an external keyboard adds a partial matching tag as well as the new tag.

I believe this is caused by this section of code:
https://github.com/laurent22/joplin/blob/4f5bbc113226fd7a63baa2587ba3480b24b28ac7/packages/app-mobile/components/ComboBox.tsx#L455C5-L462C4

Reproduction steps (using portrait orientation on Android emulator):
1. Create a tag on a note called 'add'
2. On another note, type 'ad' in the search box of the tag association screen
3. Press enter
4. Observe that the 'add' tag has been added, in addition to a new tag called 'ad'

### Expected behaviour

Only the new tag should be added in this scenario
